pinot-common/src/main/java/org/apache/pinot/common/function/DateTimeFunctions.java

```diff
@@ -241,6 +241,15 @@ static Long fromDateTime(String dateTimeString, String pattern) {
     return DateTimePatternHandler.parseDateTimeStringToEpochMillis(dateTimeString, pattern);
   }
 
+
+  /**
+   * Round the given time value to nearest rounding bucket
+   */
+  @ScalarFunction
+  static Long round(Long timeValue, Number roundToNearest) {
+    return (timeValue / roundToNearest.intValue()) * roundToNearest.intValue();
```

Comment:
Use `roundToNearest.longValue()` and cache it for performance concern.

Reply:
We apply this on the stream, which presumably has different input values coming in (e.g. millis). Not sure how we can cache it?

Reply:
But using long is better, yes.

Reply:
changed to roundToNearest.longValue and cached it